src/components/base/outline-alert/outline-alert.stories.ts

```diff
@@ -0,0 +1,158 @@
+import { html, TemplateResult } from 'lit';
+import './outline-alert';
+import { argTypeSlotContent } from '../../base/outline-element/utils/utils';
+import { alertSizes, alertStatusTypes } from './outline-alert';
+import { ifDefined } from 'lit/directives/if-defined';
+
+export default {
+  title: 'Molecules/Alert',
+  component: 'outline-alert',
+  argTypes: {
+    headerSlot: {
+      name: 'slot="outline-alert--header"',
+      description: 'The header slot. For example a header title.',
+      table: {
+        category: 'Slots',
+      },
+      // We aren't setting a control type here so we can edit the value using the infered object.
+    },
+    // We aren't setting a control type here so we can edit the value using the infered object.
+    defaultSlot: {
+      ...argTypeSlotContent.defaultSlot,
+      table: {
+        category: 'Slots',
+      },
+    },
+    statusType: {
+      description:
+        'The status type of the alert. Such as `information` or `warning`.',
+      options: alertStatusTypes,
+      control: { type: 'select' },
+    },
+    size: {
+      description: 'The size of the alert.',
+      options: alertSizes,
+      control: { type: 'select' },
+    },
+    shouldShowIcon: {
+      description: '(not yet implemented) Should we show the icon',
+      control: { type: 'boolean' },
+    },
+    isInteractive: {
+      description: 'Is there an interaction in the alert, such as a button.',
+      control: { type: 'boolean' },
+    },
+  },
+  parameters: {
+    docs: {
+      description: {
+        component: `
+This component renders an alert.
+
+## Variation
+
+You can set the type of alert with \`statusType\`.
+
+You can use a smaller alert with \`size\` of \`small\`.
+
+You can remove the icon with \`shouldShowIcon\` set to \`false\`.
+
+You can customize the header by adding a \`outline-alert--header\` slot.
+
+## Accessibility
+
+If the alert has an interaction, you should indicate this with \`isInteractive\` set to \`true\`.
+`,
+      },
+      source: {
+        // This code sample will be used for every example unless overridden.
+        code: `
+<outline-alert
+  statusType="{{ statusType }}"
+  size="{{ size }}"
+  {{ shouldShowIcon }}
+  {{ isInteractive }}
+>
+  <span slot="outline-alert--header">{{ headerSlot }}</span>
+  {{ defaultSlot }}
+</outline-alert>
+        `,
+      },
+    },
+  },
+};
+
+const Template = ({
+  headerSlot,
+  defaultSlot,
+  statusType,
+  size,
+  shouldShowIcon,
+  isInteractive,
+}): TemplateResult => {
+  return html`
+    <outline-alert
+      statusType="${ifDefined(statusType)}"
+      size="${ifDefined(size)}"
+      ?shouldShowIcon="${shouldShowIcon}"
+      ?isInteractive="${isInteractive}"
+    >
+      ${ifDefined(headerSlot)} ${ifDefined(defaultSlot)}
+    </outline-alert>
+  `;
+};
+
+export const Information = Template.bind({});
```

Comment:
I don't think we need a SB example for every possible iteration/variant.

Reply:
I think it's beneficial to write stories for every visual representation of a component like we do with the button component, especially as we implement VRT. It would be good to detect if something you did broke a layout of a component in a specific state.

Reply:
I'm down to have this discussion in a larger setting, but for me SB serves t purposes. 1: To show what the component can do. 2: To show how to use the component. 
I think 1 is taken care of by the controls we take the time to set up. So the number of variants in a story should be dictated by the different ways the component can be used. EX: the alertdialog option here.

Reply:
@LeeMellon , I think that's an interesting point.

This is the first time I've used storybook, so the patterns are new to me. I can see the point of view that a smaller number of stories may be easier for an implementer since it focuses their attention.

If it's helpful, this is what I've used Storybook for so far.

I write the storybook page first with examples of the variations that I understand need to be represented. In the alert, this means one of each status type. There is some level of feeling things out here. On a current project, we have an icon that can show 1800 icons and I don't have 1800 stories. So, I think 4 statuses makes sense to me as stories, but 1800 doesn't. I made another component to show all the colors / icons and those have their own component pages. In development, I was settings CSS classes per status type, but there were no CSS changes for those icons. Meaning, it was possible for me to break a single status type, but not a single icon.

If there is JS interaction, I may write the Jest tests at this point too.

Then, I start implementing the component itself. At this time, I'm watching those stories to see the components change and gradually get the features that I expect. Most of the time I'm looking at the `docs` pane, or for interaction / screen reader the `iframe` example.

If I forgot an example, I'll add more stories to capture those use cases.

... And I don't really use the controls much.

I agree with @mabry1985, that for visual regression testing, some of the additional variations may be helpful since the tools will default to one screenshot of each.

Might be a good team discussion or one for Github discussions.